scripts/travis-pr-title-lint

```diff
@@ -0,0 +1,181 @@
+
+const shelljs = require('shelljs');
+
+const {
+  TRAVIS_REPO_SLUG,
+  TRAVIS_PULL_REQUEST,
+  VERBOSE,
+} = process.env;
+
+
+const DONT_PANIC_MESSAGE = `
+Don't panic! If your travis build is failing here, please take a look at
+
+ - https://github.com/mozilla/web-ext/blob/master/CONTRIBUTING.md#writing-commit-messages
+
+and feel free to ask for help from one of the maintainers in a comment; we are here to help ;-)
+`;
+
+function findMergeBase() {
+  const res = shelljs.exec('git merge-base HEAD master', {silent: true});
+
+  if (res.code !== 0) {
+    throw new Error(`findMergeBase Error: ${res.stderr}`);
+  }
+
+  const baseCommit = res.stdout.trim();
+  if (VERBOSE === 'true') {
+    console.log('DEBUG findMergeBase:', baseCommit);
+  }
+
+  return baseCommit;
+}
+
+function getGitBranchCommits() {
+  const baseCommit = findMergeBase();
+  const gitCommand = `git rev-list --no-merges HEAD ^${baseCommit}`;
+
+  const res = shelljs.exec(gitCommand, {silent: true});
+
+  if (res.code !== 0) {
+    throw new Error(`getGitBranchCommits Error: ${res.stderr}`);
+  }
+
+  const commits = res.stdout.trim().split('\n');
+  if (VERBOSE === 'true') {
+    console.log('DEBUG getGitBranchCommits:', commits);
+  }
+
+  return commits;
+}
+
+function getGitCommitMessage(commitSha1) {
+  const res = shelljs.exec(`git show -s --format=%B ${commitSha1}`, {silent: true});
+
+  if (res.code !== 0) {
+    throw new Error(`getGitCommitMessage Error: ${res.stderr}`);
+  }
+
+  const commitMessage = res.stdout.trim();
+  if (VERBOSE === 'true') {
+    console.log(`DEBUG getGitCommitMessage: "${commitMessage}"`);
+  }
+
+  return commitMessage;
+}
+
+function getPullRequestTitle() {
+  return new Promise(function(resolve, reject) {
+    const pullRequestURLPath = `/${TRAVIS_REPO_SLUG}/pull/${TRAVIS_PULL_REQUEST}`;
+
+    console.log(`Retrieving the pull request title from https://github.com${pullRequestURLPath}\n`);
+
+    var req = https.get({
+      host: 'github.com',
+      path: pullRequestURLPath,
+      headers: {
+        'User-Agent': 'GitHub... your API can be very annoying ;-)',
+      },
+    }, function(response) {
+      if (response.statusCode < 200 || response.statusCode > 299) {
+        reject(new Error(`getPullRequestTitle got an unexpected statusCode: ${response.statusCode}`));
+        return;
+      }
+
+      var body = '';
+      response.on('data', function(data) {
+        try {
+          body += data;
+
+          // Once we get the closing title tag, we can read the pull request title and
+          // close the http request.
+          if (body.includes('</title>')) {
+            response.removeAllListeners('data');
+            response.emit('end');
+
+            var titleStart = body.indexOf('<title>');
+            var titleEnd = body.indexOf('</title>');
+
+            // NOTE: page slice is going to be something like:
+            // "<title> PR title by author · Pull Request #NUM · mozilla/web-ext · GitHub"
```

Comment:
The logic here does not account for HTML entities. Could that ever cause problems?

It seems that `"`, `'` and `>` are escaped, and I expect that `&` and `<` would also be escaped, at the least. For example, see issues 813, 1310, 1107

Reply:
Well, theoretically it may not be causing any problem, but for sure we will be linting something different from the actual message, and so I opted to add [html-entities](https://github.com/mdevils/node-html-entities) to the dev dependencies and use it to decode the message retrieved from the webpage before resolving it.